scripts/generate_cpp_code.py

```diff
@@ -0,0 +1,781 @@
+                [
+                    f'error = {item.name}{item.dereference_style}TryFromJSON(obj);',
+                    'if (error.empty()) {',
+                    f'\thas_{item.name} = true;',
+                    '}',
+                ]
+            )
+        res.extend(
+            [
+                f'if ({condition}) {{'
+                f'\treturn "{self.name} failed to parse, none of the anyOf candidates matched";'
+                '}'
+            ]
+        )
+        return res
+
+    def write_nested_classes_header(self) -> List[str]:
+        if not self.nested_classes:
+            return []
+        res = []
+        for item, nested_class in self.nested_classes.items():
+            res.extend(nested_class.write_header())
+        return [f'\t{x}' for x in res]
+
+    def write_nested_classes_source(self, base_class: List[str]) -> List[str]:
+        if not self.nested_classes:
+            return []
+        res = []
+        for item, nested_class in self.nested_classes.items():
+            new_base_class = []
+            new_base_class.extend(base_class)
+            new_base_class.append(self.name)
+            res.extend(nested_class.write_source(new_base_class))
+        return [f'{x}' for x in res]
+
+    def write_variables(self) -> List[str]:
+        res = []
+        if self.variables:
+            res.append('public:')
+            res.extend(self.variables)
+        return res
+
+    def write_source(self, base_class: List[str]) -> List[str]:
+        res = []
+        base = ''
+        if base_class:
+            base = '::'.join(base_class) + '::'
+        res.extend([f'{base}{self.name}::{self.name}() {{}}'])
+        res.extend(self.write_nested_classes_source(base_class))
+        res.extend(
+            [
+                '',
+                f'{base}{self.name} {base}{self.name}::FromJSON(yyjson_val *obj) {{',
+                f'\t{self.name} res;',
+                '\tauto error = res.TryFromJSON(obj);',
+                '\tif (!error.empty()) {',
+                '\t\tthrow InvalidInputException(error);',
+                '\t}',
+                '\treturn res;',
+                '}',
+                '',
+                f'string {base}{self.name}::TryFromJSON(yyjson_val *obj) {{',
+                '\tstring error;',
+            ]
+        )
+        res.extend(self.write_one_of())
+        res.extend(self.write_all_of())
+        res.extend(self.write_any_of())
+        res.extend(self.try_from_json_body)
+        res.extend(['\treturn string();', '}'])
+        return res
+
+    def write_header(self) -> List[str]:
+        res = []
+        res.extend([f'class {self.name} {{', 'public:', f'\t{self.name}();'])
+        res.extend(self.write_nested_classes_header())
+        res.extend(
+            [
+                'public:',
+                f'\tstatic {self.name} FromJSON(yyjson_val *obj);',
+                'public:',
+                '\tstring TryFromJSON(yyjson_val *obj);',
```

Comment:
Just a thought, maybe (as future work) we want to create a stack of errors:
```c++
bool TryFromJSON(yyjson_val *obj, stack<string> &error);
```

Especially since we have these `Object[0-9]+` classes, these errors aren't very descriptive if they happen at deeper levels.
With a stack we can use:
```
if (!TryFromJSON(val, error)) {
    error.push("<ClassName> failed to parse");
    return false;
}
```